xmtp_mls/src/groups/mod.rs

```diff
@@ -582,108 +582,87 @@ impl<ScopedClient: ScopedGroupClient> MlsGroup<ScopedClient> {
 
     // Create a group from a decrypted and decoded welcome message
     // If the group already exists in the store, overwrite the MLS state and do not update the group entry
-    async fn create_from_welcome(
+
+    pub(super) async fn create_from_welcome(
         client: &ScopedClient,
         provider: &XmtpOpenMlsProvider,
         welcome: MlsWelcome,
         added_by_inbox: String,
         welcome_id: i64,
+        cursor: Option<i64>,
     ) -> Result<Self, GroupError>
     where
         ScopedClient: Clone,
     {
         tracing::info!("Creating from welcome");
-        let mls_welcome =
+        let staged_welcome =
             StagedWelcome::new_from_welcome(provider, &build_group_join_config(), welcome, None)?;
```

Comment:
This is where we re-form the staged welcome from the decrypted welcome parts. Maybe we could just pass the decrypted welcome directly, and store the staged welcome on the decrypted welcome?